### PR TITLE
[MULTIARCH-1073, 1074] SBO updates for OCP 4.9 from P and Z 

### PR DESCRIPTION
**Pull request by ktania46**
Jiras:
for P : https://issues.redhat.com/browse/MULTIARCH-1073
for Z: https://issues.redhat.com/browse/MULTIARCH-1074

Drafts : 
1. https://docs.google.com/document/d/1sfJZnMY3W7gihxF8-BGa1FiegpSkOgdmf0W8DRhETHk/edit#  
2. https://docs.google.com/document/d/1m-aRuCtmybcwzzB6TDQdoQ2ImSE-yP9iDmjMcyacm_0/edit?pli=1#
3. https://docs.google.com/document/d/1sfJZnMY3W7gihxF8-BGa1FiegpSkOgdmf0W8DRhETHk/edit#heading=h.pp8h16551m20

Preview: https://deploy-preview-37964--osdocs.netlify.app/openshift-enterprise/latest/applications/connecting_applications_to_services/getting-started-with-service-binding-ibm-power-ibm-z

Reviewers :
For P : Pravin D-Silva
for Z : Johanna Husta

cc @SNiemann15

### CODE REVIEW

**Review by codyhoag — COMMENTED** [review]
This is looking good! I've made some suggestions on formatting and general step setup. Let me know if you have any questions.

**Comment by johusta** [comment]
For the first example on checking after you install the PostgreSQL operator your have output for creating a db. If you want out put after the first executed yaml I am guessing you may want to check if the ibm-multiarch-catalog pod is running? 

```

[root@ospbmgr6 SBO-example]# oc get pods -n openshift-marketplace
NAME                                                              READY   STATUS      RESTARTS      AGE
0fe8ebf1dfcfa5ca70ac59c230368d2afbf0ad9ba6146049d3eae3--1-7shlt   0/1     Completed   0             106m
certified-operators-z6qft                                         1/1     Running     0             18h
community-operators-sl6td                                         1/1     Running     0             61m
ibm-multiarch-catalog-dfbch                                       1/1     Running     0             100m
marketplace-operator-b68bcb6dc-lqd8p                              1/1     Running     1 (18h ago)   19h
redhat-marketplace-kk9ff                                          1/1     Running     0             18h
redhat-operators-429vw                                            1/1     Running     0             29m

**Comment by johusta** [comment]
The example check of oc get pods -n my-postgresql should be used after Creating a PostgreSQL database instance. The current one you have has an additional pod running which is no longer correct. So you want the following:

```

[root@ospbmgr6 SBO-example]# oc get pods -n my-postgresql
NAME                             READY   STATUS    RESTARTS   AGE
sampledatabase-cbc655488-kxpx6   0/1     Running   0          83s

File: modules/sbo-deploying-the-spring-petclinic-sample-application-power-z.adoc [hunk 1]
```diff
@@ -0,0 +1,67 @@
+[id="sbo-deploying-the-spring-petclinic-sample-application-ibm-power-z_{context}"]
+= Deploying the Spring PetClinic sample application
+
+To deploy the Spring PetClinic sample application on an {product-title} cluster, you must use a deployment configuration and configure your local environment to test the application.
+
+.Procedure
+
+. Switch to the *Developer* perspective and ensure that you are in the appropriate project, for example, `my-postgresql`. Click the btn:[+Add] button and select *Import from Git*.
```
Comment (johusta) [inline modules/sbo-deploying-the-spring-petclinic-sample-application-power-z.adoc:8]: Being we now have a multi arch image we can use the x86 steps under deploying the spring pet clinic and replace the p/z section with those steps.  Required changes will be Under step 1 we need to change the yaml

1. Deploy the spring-petclinic-rest application with the PostgresCluster custom resource (CR) by running the following command in shell:

```

oc apply -f - << EOD
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: spring-petclinic-rest
  namespace: my-postgresql
  labels:
    app: spring-petclinic-rest
spec:
  replicas: 1
  selector:
    matchLabels:
      app: spring-petclinic-rest
  template:
    metadata:
      labels:
        app: spring-petclinic-rest
    spec:
      containers:
        - name: application
          image: quay.io/service-binding/spring-petclinic-rest:latest
          env:
          - name: SPRING_PROFILES_ACTIVE
            value: postgresql,spring-data-jpa
          - name: org.springframework.cloud.bindings.boot.enable
            value: "true"
          ports:
          - name: http
            containerPort: 9966
      imagePullPolicy: Always
---
apiVersion: v1
kind: Service
metadata:
  name: spring-petclinic-rest
  namespace: my-postgresql
spec:
  ports:
  - port: 80
    targetPort: 9966
  selector:
    app: spring-petclinic-rest
---
EOD

```

File: modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+To connect the sample application to the database service, you must create a `ServiceBinding` custom resource (CR) that triggers the {servicebinding-title} to project the binding data into the application.
+
+.Procedure
+
+. Create a `ServiceBinding` CR to project the binding data:
++
+[source,terminal]
+----
+$ oc apply -f - << EOD
+---
```
Comment (johusta) [inline modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc:13]: Replace the yaml example to the following:

```

oc apply -f - << EOD
---
apiVersion: binding.operators.coreos.com/v1alpha1
kind: ServiceBinding
metadata:
    name: sb-spring-petclinic-rest
    namespace: my-postgresql
spec:
    services: 
    - group: postgresql.dev4devs.com
      version: v1alpha1
      kind: Database 
      name: sampledatabase
    application: 
      name: spring-petclinic-rest
      group: apps
      version: v1
      resource: deployments
---
EOD

```

Comment (johusta) [inline modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc:13]: Replace the yaml example to the following:

```

oc apply -f - << EOD
---
apiVersion: binding.operators.coreos.com/v1alpha1
kind: ServiceBinding
metadata:
    name: sb-spring-petclinic-rest
    namespace: my-postgresql
spec:
    services: 
    - group: postgresql.dev4devs.com
      version: v1alpha1
      kind: Database 
      name: sampledatabase
    application: 
      name: spring-petclinic-rest
      group: apps
      version: v1
      resource: deployments
---
EOD

```

File: modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+apiVersion: binding.operators.coreos.com/v1alpha1
+kind: ServiceBinding
+metadata:
+    name: spring-petclinic-rest
+    namespace: my-postgresql
+spec:
+    services: <1>
+    - group: postgresql.dev4devs.com
+      version: v1beta1
```
Comment (johusta) [inline modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc:22]: version is incorrect

```
version:  v1alpha1

```

Comment (johusta) [inline modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc:22]: version is incorrect

```
version:  v1alpha1

```

Comment (johusta) [inline modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc:22]: version is incorrect

```
version:  v1alpha1

```

Comment (johusta) [inline modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc:22]: version is incorrect

```
version: v1alpha1

```

File: modules/sbo-deploying-a-postgresql-database-operator-power-z.adoc [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+EOD
+----
+<1> The Operator image. 
+* For IBM Power Systems: `quay.io/ibm/operator-registry-ppc64le:release-4.9` 
+* For IBM Z and LinuxONE: `quay.io/ibm/operator-registry-s390x:release-4.8`
+
+.Example output
+[source,terminal]
+----
+$ oc get pods -n my-postgresql
```
Comment (johusta) [inline modules/sbo-deploying-a-postgresql-database-operator-power-z.adoc:70]: This is the wrong check for the operator. Should be changed to check the subscriptions under openshift-operators:

```
[root@ospbmgr6 SBO-x86]# oc get subs -n openshift-operators
NAME                               PACKAGE                            SOURCE                  CHANNEL
postgresql-operator-dev4devs-com   postgresql-operator-dev4devs-com   ibm-multiarch-catalog   alpha
rh-service-binding-operator        rh-service-binding-operator        redhat-operators        stable
```

File: modules/sbo-creating-a-postgresql-database-instance-power-z.adoc [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+$ oc get pods -n my-postgresql
+----
++
+The output verifies that the database is created:
++
+.Example output
+[source,terminal]
+----
+NAME                                     READY    STATUS      RESTARTS   AGE
+postgresql-operator-56bd6964c4-j2slb      0/1     Running     0          1m
```
Comment (johusta) [inline modules/sbo-creating-a-postgresql-database-instance-power-z.adoc:69]: Get rid of postgresql-operator from the example output it should only be the sample database eg:

```
[root@ospbmgr6 SBO-x86]# oc get pods -n my-postgresql
NAME                             READY   STATUS    RESTARTS   AGE
sampledatabase-cbc655488-74kss   0/1     Running   0          32s

```

File: modules/sbo-deploying-the-spring-petclinic-sample-application-power-z.adoc [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
++
+[source,text]
+.Example output
+----
+NAME                                     READY    STATUS      RESTARTS   AGE
+postgresql-operator-56bd6964c4-j2slb     1/1     Running       0        91m
+sampledatabase-cbc655488-xjlxt           1/1     Running       0        81m
+spring-petclinic-rest-1-build            0/1     Completed     0        10m
+spring-petclinic-rest-7bdbdc987b-rfvnq   1/1     Running       0        4m48s
+----
```
Comment (johusta) [inline modules/sbo-deploying-the-spring-petclinic-sample-application-power-z.adoc:47]: The new output to check if the spring pet clinic is running will now be 

```
oc get pods -n my-postgresql
NAME                                     READY   STATUS    RESTARTS   AGE
sampledatabase-cbc655488-6n25f           1/1     Running   0          8m14s
spring-petclinic-rest-7659d5b774-zkjpg   1/1     Running   0          3m30s

```

File: modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+      "status": "True",
+      "type": "Ready",
+    },
+  ],
+]
+----
++
+[source,terminal]
+----
+$ oc get pods
```
Comment (johusta) [inline modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc:84]: Don't need the oc get pods after the Service Binding is done.

File: modules/sbo-deploying-the-spring-petclinic-sample-application-power-z.adoc (hunk 1 above)
Comment (pedjak) [reply modules/sbo-deploying-the-spring-petclinic-sample-application-power-z.adoc:8]: @johusta was this new image tested on p/z? can you confirm please?

Comment (johusta) [reply modules/sbo-deploying-the-spring-petclinic-sample-application-power-z.adoc:8]: @pedjak yeah I have tested that a couple of times with that new multi arch image on Z. @pravin-dsilva is verifying on P. @ktania46 you mentioned you wanted to change the x86 docs to use this image as well. Someone should test on x86 to make sure it works.

**Comment by ktania46** [comment]
This PR has been closed due to rebase conflicts fresh PR for same https://github.com/openshift/openshift-docs/pull/39357

**Pull request closed without merging**